### PR TITLE
Migrate the OSSInsight API to public endpoint

### LINKED ISSUE #522: Migrate the OSSInsight API to public endpoint
Earlier, OSSInsight officially released the [Public API](https://ossinsight.io/docs/api). Now we can migrate the API endpoint used in Github 1s to the new public endpoint.

Document:

https://ossinsight.io/docs/api/list-trending-repos

Example:

```shell
curl -L -X GET 'https://api.ossinsight.io/v1/trends/repos/?period=past_24_hours&language=All' \
-H 'Accept: application/json'
```

### PR DESCRIPTION
**Pull request by conwnet**
This PR resolves #522

### CODE REVIEW

**Review by Mini256 — COMMENTED** [review]
Is there any problem hinder the migration process? Is there anything I can help with.

I tried it out in the preview environment and found that:

- Need to using `TypeScript` instead of `Typescript`: `https://api.ossinsight.io/v1/trends/repos/?language=TypeScript&period=past_week`
- Collections Page seems broken in the preview

**Comment by conwnet** [comment]
@Mini256 Thank you so much for pointing out the problem. I have already fixed it.

Is there a way to get all the Collections in the new Public API? I'm still using the old API here, Is there any plan to remove this Collections API?

https://api.ossinsight.io/collections

**Comment by Mini256** [comment]
@conwnet Yes, we have added a new API: https://ossinsight.io/docs/api/list-collections/

File: extensions/github1s/src/adapters/ossinsight/interfaces.ts [hunk 1]
```diff
@@ -12,104 +12,82 @@ const resolveDataFromResponse = (response: Response) => {
...
-	return response.json().then((body) => body?.data || []);
+	return response.json().then((body) => body?.data?.rows || []);
 };
 
 export type RepoItem = {
 	collection_names?: string;
 	contributor_logins?: string;
 	description?: string;
-	forks?: number;
+	forks: string;
```
Comment (Siddhant-K-code) [inline extensions/github1s/src/adapters/ossinsight/interfaces.ts:22]: Any reason to remove it from being an optional entity?

```suggestion
	forks?: string;
```

**Review by Siddhant-K-code — APPROVED** [review]
Looks great to me 🎉

File: extensions/github1s/src/adapters/ossinsight/interfaces.ts (hunk 1 above)
Comment (conwnet) [reply extensions/github1s/src/adapters/ossinsight/interfaces.ts:22]: It is marked as required in the api document of ossinsight

**Pull request merged**